jbmc/src/java_bytecode/jar_file.h

```diff
@@ -13,8 +13,6 @@ Author: Diffblue Ltd
 #include <string>
 #include <vector>
 
-#include <util/optional.h>
```

Comment:
It seemed to work fine on my Ubuntu 20.04, but other OS/compiler combinations indeed make those includes necessary. I had naively assumed that standard library headers would likely (transitively) include that anyway, but as experimenting shows that's not always the case. So, yes, I may well end up putting this back in.

Reply:
Good to see this being fixed.

I'd prefer to avoid relying on transitive header includes where possible. With `optional` as an example my reasoning goes something like -

1. Header "A" depends on `optional` so `#include <optional>` is added to it.
2. `.cpp` file "B" depends on header "A" and uses `std::optional` in its implementation. It therefore depends on `optional`, but as this `.cpp` already includes header "A" `#include <optional>` is not added to it.
3. Maintainer of header "A" decides they no longer need to depend on `optional`. Lets say they were using it for the type of a private member function rather than as part of the exposed API. They remove the `#include <optional>` from header "A", because it no longer needs it itself.
4. The compilation of `.cpp` "B" now fails. This means that the maintainer of header "A" now needs to either add the include to the `.cpp` file, or to leave the extra include in their header.

Leaving the extra include in could potentially extend compile times. Adding it to the `.cpp` increases the number of changes in their PR and potentially requires approval from more code owners. The decision and additional work could have been avoided if the reliance on transitive includes had been avoided.